util/src/main/scala/org/broadinstitute/dsde/rawls/util/Retry.scala

```diff
@@ -16,21 +16,29 @@ trait Retry {
   val system: ActorSystem
 
   type Predicate[A] = A => Boolean
-  type RetryCountConsumer[A] = (A, Int) => A
 
-  def always[A]: Predicate[A] = _ => true
+  /**
+    * A Future that has potentially been retried, with accumulated errors.
+    * There are 3 cases:
+    * 1. The future failed 1 or more times, and the final result is an error.
+    *   - This is represented as {{{Left(NonEmptyList(errors))}}}
+    * 2. The future failed 1 or more times, but eventually succeeded.
+    *   - This is represented as {{{Right(List(errors), A)}}}
+    * 3. The future succeeded the first time.
+    *   - This is represented as {{{Right(List.empty, A)}}}
+    */
+  type AccumulatingFuture[A] = Future[Either[NonEmptyList[Throwable], (List[Throwable], A)]]
```

Comment:
Sure maybe `RetryableFuture`? This behavior is very specific to retries, so it makes sense to name it as such